gensrc/proto/segment.proto

```diff
@@ -190,6 +190,8 @@ message ColumnMetaPB {
     optional uint64 total_mem_footprint = 31;
     // for json column only
     optional JsonMetaPB json_meta = 32;
+    // for json flat column only
+    optional bytes name = 33;
```

Comment:
why not put it into `json_meta`

Reply:
the column type after flattening JSON may not be JSON type in the future, so I wasn't put it into `json_meta`, but it's a good idea

Reply:
I remove the change in this PR, and reconsider it in next BE's PR